src/lib/address/address.ts

```diff
@@ -1,496 +1,79 @@
-        ) {
-          const tokenAssetId = toAssetId({
-            chainId,
-            assetNamespace: ASSET_NAMESPACE.erc20,
-            assetReference: parsedUrl.target_address.toLowerCase(),
-          })
-
-          const asset = selectAssetById(store.getState(), tokenAssetId)
-          if (!asset) throw new Error(DANGEROUS_ETH_URL_ERROR)
-
-          const amountCryptoPrecision = parsedUrl.parameters.uint256
-            ? fromBaseUnit(parsedUrl.parameters.uint256, asset.precision)
-            : undefined
-
-          return {
-            assetId: tokenAssetId,
-            maybeAddress: parsedUrl.parameters.address,
-            chainId,
-            ...(amountCryptoPrecision && { amountCryptoPrecision }),
-          }
-        }
-
-        // Native EVM asset transfers
-        const assetId = inputAssetId || getChainAdapterManager().get(chainId)?.getFeeAssetId()
 
-        const rawAmount = parsedUrl.parameters?.value ?? parsedUrl.parameters?.amount
-        const asset = selectAssetById(store.getState(), assetId ?? '')
-        const amountCryptoPrecision =
-          rawAmount && asset ? fromBaseUnit(rawAmount, asset.precision) : undefined
-
-        return {
-          assetId,
-          maybeAddress: parsedUrl.target_address ?? urlOrAddress,
-          chainId,
-          ...(amountCryptoPrecision && { amountCryptoPrecision }),
-        }
-      } catch (error) {
-        if (error instanceof Error) {
-          if (error.message === DANGEROUS_ETH_URL_ERROR) throw error
-          // address, not url, don't log
-          if (error.message.includes('Not an Ethereum URI')) break
-        }
-        console.error(error)
-      }
-      break
-    case btcChainId:
-    case bchChainId:
-    case dogeChainId:
-    case ltcChainId:
-      try {
-        const urnScheme = CHAIN_ID_TO_URN_SCHEME[inputChainId]
-        const parsedUrl = bip21.decode(urlOrAddress, urnScheme)
-        return {
-          assetId: inputAssetId,
-          maybeAddress: parsedUrl.address,
-          chainId: inputChainId,
-          ...(parsedUrl.options?.amount
-            ? { amountCryptoPrecision: bnOrZero(parsedUrl.options.amount).toFixed() }
-            : {}),
-        }
-      } catch (error) {
-        if (error instanceof Error) {
-          // address, not url, don't log
-          if (error.message.includes('Invalid BIP21 URI')) break
-        }
-        console.error(error)
-      }
-      break
-    default:
-      return { assetId: inputAssetId, chainId: inputChainId, maybeAddress: urlOrAddress }
-  }
-
-  return { assetId: inputAssetId, chainId: inputChainId, maybeAddress: urlOrAddress }
-}
-
-export const parseMaybeUrl = async ({
-  urlOrAddress,
+export type {
+  ParseAddressInputReturn,
+  ResolveVanityAddress,
+  ResolveVanityAddressReturn,
+  ReverseLookupVanityAddress,
+  ValidateVanityAddress,
+} from './types'
+export { validateAddress } from './validation'
+
+export const parseAddress = async ({
```

Comment:
Now does just that - address parsing, no URL shenanigans. Same exhausting of ChainIds until obtaining a ChainId for the address and a fee AssetId, that didn't change.